lib/src/migrators/module/references.dart

```diff
@@ -0,0 +1,326 @@
+
+  /// Maps canonical stylesheet URLs to the parsed stylesheet used when building
+  /// this object.
+  ///
+  /// Users of this object should get stylesheets from this object instead of
+  /// re-parsing them.
+  Map<Uri, Stylesheet> get stylesheets => UnmodifiableMapView(_stylesheets);
+
+  /// Returns true if the member declared by [declaration] is referenced within
+  /// another stylesheet.
+  bool referencedOutsideDeclaringStylesheet(SassNode declaration) {
+    Set<SassNode> references;
+    if (declaration is FunctionRule) {
+      references = functions
+          .keysForValue(declaration)
+          .union(getFunctionReferences.keysForValue(declaration));
+    } else if (declaration is MixinRule) {
+      references = mixins.keysForValue(declaration);
+    } else {
+      references = variables.keysForValue(declaration);
+    }
+    return references.any(
+        (reference) => reference.span.sourceUrl != declaration.span.sourceUrl);
+  }
+
+  /// Finds the original declaration of the variable referenced in [reference].
+  SassNode /*VariableDeclaration|Argument*/ originalDeclaration(
+      VariableExpression reference) {
+    var declaration = variables[reference];
+    return variableReassignments[declaration] ?? declaration;
+  }
+
+  /// For a given declaration or reference, returns the set of all nodes
+  /// associated with it, including itself.
+  ///
+  /// For mixins, this is the declaring MixinRule and all referencing
+  /// IncludeRules.
+  ///
+  /// For functions, this is the declaring FunctionRule and all referencing
+  /// FunctionRules (both regular and statically-determined get-function calls).
+  ///
+  /// For variables, this is the original declaration, all reassignments, and
+  /// all referencing VariableExpressions.
+  Set<SassNode> allNodes(SassNode node) {
+    if (node is IncludeRule) return allNodes(mixins[node]);
+    if (node is FunctionExpression) return allNodes(functions[node]);
+    if (node is VariableExpression) return allNodes(originalDeclaration(node));
+    if (node is MixinRule) return {node, ...mixins.keysForValue(node)};
+    if (node is FunctionRule) {
+      return {
+        node,
+        ...functions.keysForValue(node),
+        ...getFunctionReferences.keysForValue(node)
+      };
+    }
+    return {
+      node,
+      ...variables.keysForValue(node),
+      for (var reassignment in variableReassignments.keysForValue(node))
+        ...variables.keysForValue(reassignment)
+    };
+  }
+
+  /// Constructs a new References object based on the stylesheet at [entrypoint]
+  /// and its dependencies.
+  static References build(Uri entrypoint) =>
+      _ReferenceVisitor().build(entrypoint);
+}
+
+/// A visitor that builds a References object.
+class _ReferenceVisitor extends RecursiveAstVisitor {
+  /// The References object being built.
+  final _references = References._();
+
+  /// The current global scope.
+  ///
+  /// This persists across imports, but not across module uses.
```

Comment:
```suggestion
  /// This persists across imports, but not across module loads.
```

This better matches the terminology I've been using in docs.